dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -286,5 +291,50 @@ try
 }
 CATCH
 
+TEST_F(ComputeServerRunner, cancelAggTasks)
+try
+{
+    startServers(4);
+    {
+        auto [start_ts, res] = injectCancel(context
+                                                .scan("test_db", "test_table_1")
+                                                .aggregation({Max(col("s1"))}, {col("s2"), col("s3")})
+                                                .project({"max(s1)"}));
+        MockComputeServerManager::instance().cancelQuery(start_ts);
+        assertQueryCancelled(start_ts);
+    }
+}
+CATCH
+
+TEST_F(ComputeServerRunner, cancelJoinTasks)
+try
+{
+    startServers(4);
+    {
+        auto [start_ts, res] = injectCancel(context
+                                                .scan("test_db", "l_table")
+                                                .join(context.scan("test_db", "r_table"), tipb::JoinType::TypeLeftOuterJoin, {col("join_c")}));
+        MockComputeServerManager::instance().cancelQuery(start_ts);
+        assertQueryCancelled(start_ts);
+    }
+}
+CATCH
+
+TEST_F(ComputeServerRunner, cancelJoinThenAggTasks)
+try
+{
+    startServers(4);
+    {
+        auto [start_ts, res] = injectCancel(context
+                                                .scan("test_db", "l_table")
+                                                .join(context.scan("test_db", "r_table"), tipb::JoinType::TypeLeftOuterJoin, {col("join_c")})
+                                                .aggregation({Max(col("l_table.s"))}, {col("l_table.s")})
+                                                .project({col("max(l_table.s)"), col("l_table.s")}));
+        MockComputeServerManager::instance().cancelQuery(start_ts);
+        assertQueryCancelled(start_ts);
```

Comment:
Maybe we add a case like
```
auto request = xxxxxx;
auto [start_ts1, _] = injectCancel(request);
auto [start_ts2, _] = injectCancel(request);
assertQueryActive(start_ts1);
assertQueryActive(start_ts2);
MockComputeServerManager::instance().cancelQuery(start_ts1);
assertQueryCancelled(start_ts1);
assertQueryActive(start_ts2);
MockComputeServerManager::instance().cancelQuery(start_ts2);
assertQueryCancelled(start_ts2);
```

Reply:
> Maybe we add a case like
> 
> ```
> auto request = xxxxxx;
> auto [start_ts1, _] = injectCancel(request);
> auto [start_ts2, _] = injectCancel(request);
> assertQueryActive(start_ts1);
> assertQueryActive(start_ts2);
> MockComputeServerManager::instance().cancelQuery(start_ts1);
> assertQueryCancelled(start_ts1);
> assertQueryActive(start_ts2);
> MockComputeServerManager::instance().cancelQuery(start_ts2);
> assertQueryCancelled(start_ts2);
> ```

wow, assertQueryActive is really useful.